### PR TITLE
WIP: tautomer class

### PR DESCRIPTION
**Pull request by wiederm**
## Description
Starting to implement a tautomer class that handles most of the logic that is associated with the actual molecules, the tautomer conversion path, the hybrid topology, solvation, etc
## Todos
  - first try for a tautomer class
     - most of the logic is hidden, but important fields are exposed
  - changed how minimiiiiiization works

## Status
- [ ] Work in progress

### CODE REVIEW

File: neutromeratio/tautomers.py [hunk 1]
```diff
@@ -5,138 +5,304 @@
...
-    -------
-    { 'donor_idx': donor, 'hydrogen_idx' : hydrogen_idx_that_moves, 'acceptor_idx' : acceptor}
+    name: str
+        the name of the small molecule
+    intial_state_mol: rdkit.Chem.Mol
+        initial state mol
+    final_state_mol: rdkit.Chem.Mol
+        final state mol
+    nr_of_conformations : int
+        nr of conformations that are calculated
```
Comment (maxentile) [inline neutromeratio/tautomers.py]: `Parameters` block should be in docstring for `__init__`

File: neutromeratio/tautomers.py [hunk 2]
```diff
@@ -5,138 +5,304 @@
...
+        """
+
+        m1 = copy.deepcopy(self.intial_state_mol)
+        m2 = copy.deepcopy(self.final_state_mol)
+        self._perform_tautomer_transformation(m1, m2)
+        self._generate_hybrid_structure(self.intial_state_ligand_atoms, None, self.intial_state_ligand_coords[0], self.intial_state_ligand_topology)
+
+    def perform_tautomer_transformation_reverse(self):
+        """
+        Performs a tautomer transformation from the initial state to the final state
```
Comment (maxentile) [inline neutromeratio/tautomers.py]: Implementation appears right, but docstring is backwards (should be `final` --> `initial`)

File: neutromeratio/tautomers.py [hunk 3]
```diff
@@ -5,138 +5,304 @@
...
+
+        min_e = 100 * unit.kilocalorie_per_mole
+        min_coordinates = None
+
+        for _ in range(500):
+            hybrid_coord = hydrogen_mover._move_hydrogen_to_acceptor_idx(ligand_coords, override=False)
+            e = energy_function.calculate_energy(hybrid_coord, lambda_value=1.0)
+            if e < min_e:
+                min_e = e
+                min_coordinates = hybrid_coord
```
Comment (maxentile) [inline neutromeratio/tautomers.py]: Is there a reason to prefer this over gradient descent or BFGS?

File: neutromeratio/ani.py [hunk 4]
```diff
@@ -325,11 +372,14 @@ def forward(self, species_coordinates):
         # species, AEVs of fully interacting system
         species, coordinates, lam = species_coordinates
         aevs = species_coordinates[:-1]
-        if self.pbc:
+        if self.box_length != 0.0:
             cell = torch.tensor(np.array([[self.box_length, 0.0, 0.0],[0.0,self.box_length,0.0],[0.0,0.0,self.box_length]]),
                                 device=self.device, dtype=torch.float)
             aevs = aevs[0], aevs[1], cell, torch.tensor([True, True, True], dtype=torch.bool, device=self.device)
 
+        # NOTE: I am not happy about this - the order at which
```
Comment (maxentile) [inline neutromeratio/ani.py]: Agreed! I think we can proceed with this, but revisit

File: neutromeratio/utils.py [hunk 5]
```diff
@@ -14,6 +14,38 @@
 
 logger = logging.getLogger(__name__)
 
+def is_quantity_close(quantity1, quantity2, rtol=1e-10, atol=0.0):
```
Comment (maxentile) [inline neutromeratio/utils.py]: Could import from openmmtools, or include a link https://github.com/choderalab/openmmtools/blob/4616b21a16ea2933ddf8a8d130c935d52d1b2b79/openmmtools/utils.py#L392

File: neutromeratio/tests/test_neutromeratio.py [hunk 6]
```diff
@@ -48,18 +47,86 @@ def test_tautomer_transformation():
...
-    tautomer_transformation = neutromeratio.get_tautomer_transformation(from_mol, to_mol)
-    ani_input = neutromeratio.from_mol_to_ani_input(from_mol, nr_of_conf=2)
-    print(tautomer_transformation)
-    atoms = ani_input['ligand_atoms']
+    t = Tautomer(name=name, intial_state_mol=from_mol, final_state_mol=to_mol)
+    t.perform_tautomer_transformation_forward()
+    
+
+
+def test_tautomer_transformation():
```
Comment (maxentile) [inline neutromeratio/tests/test_neutromeratio.py]: Nice!
